ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -541,32 +553,43 @@ def run_ocs_upgrade(operation=None, *operation_args, **operation_kwargs):
             upgrade=True
         )
         log.info(f"Disconnected upgrade - new image: {upgrade_ocs.ocs_registry_image}")
+
+    channel = upgrade_ocs.set_upgrade_channel()
     with CephHealthMonitor(ceph_cluster):
-        channel = upgrade_ocs.set_upgrade_channel()
         upgrade_ocs.set_upgrade_images()
-        if upgrade_version != "4.9":
-            # In the case of upgrade to ODF 4.9, the ODF operator should upgrade
-            # OCS automatically.
-            upgrade_ocs.update_subscription(channel)
-        if original_ocs_version == "4.8" and upgrade_version == "4.9":
-            deployment = Deployment()
-            deployment.subscribe_ocs()
-        if (config.ENV_DATA["platform"] == constants.IBMCLOUD_PLATFORM) and not (
-            upgrade_in_current_source
-        ):
-            create_ocs_secret(config.ENV_DATA["cluster_namespace"])
-            for attempt in range(2):
-                # We need to do it twice, because some of the SA are updated
-                # after the first load of OCS pod after upgrade. So we need to
-                # link updated SA again.
-                log.info(
-                    f"Sleep 1 minute before attempt: {attempt+1}/2 "
-                    "of linking secret/SAs"
-                )
-                time.sleep(60)
-                link_all_sa_and_secret_and_delete_pods(
-                    constants.OCS_SECRET, config.ENV_DATA["cluster_namespace"]
-                )
+        if config.UPGRADE.get("ui_upgrade"):
+            ocp_version = get_ocp_version()
+
+            if (
+                Version.coerce(ocp_version) == Version.coerce("4.9")
+                and original_ocs_version == "4.8"
+                and upgrade_version == "4.9"
+            ):
+                ocs_odf_upgrade_ui()
+        else:
```

Comment:
If someone will run this on different combination than this one you will basically do not run upgrade at all and user will even get any input.
So for such case I prefer to change the logic here little bit different. 
```suggestion
        ui_upgrade_supported = False
        if config.UPGRADE.get("ui_upgrade"):
            ocp_version = get_ocp_version()

            if (
                Version.coerce(ocp_version) == Version.coerce("4.9")
                and original_ocs_version == "4.8"
                and upgrade_version == "4.9"
            ):
                ui_upgrade_supported = True
            # you can add more combinations later on here or update the condition above.
            log.warning("UI upgrade combination is not supported. It will fallback to CLI upgrade")
        if ui_upgrade_supported:
            ocs_odf_upgrade_ui()
        else:
```

Other approach will be to raise an exception that UI upgrade is not supported after the line 568 but indented out of the if block. But I think that better is to fallback to CLI without spending time on all pre upgrade tests and then fail with exception to upgrade.

Reply:
@petr-balogh Its better to raise exception instead of falling back to CLI for upgrade if not supported, because user is expecting upgrade through UI and the results are misleading if we do CLI upgrade.

Reply:
@vavuthu the same approach we use in UI deployment. If UI deployment is not implemented we fall back to CLI. If we already have the same approach in deployment, I think we should follow the same in upgrade.

Reply:
synced with @petr-balogh and agreed with falling back to CLI